### PR TITLE
Add set_normalized_coefficients

### LINKED ISSUE #1183: Support MOI modifications
http://www.juliaopt.org/MathOptInterface.jl/latest/apireference.html#Functions-and-function-modifications-1

### PR DESCRIPTION
**Pull request by odow**
Closes #1183

### CODE REVIEW

File: src/constraints.jl [hunk 1]
```diff
@@ -682,6 +682,46 @@ function set_normalized_coefficient(
...
 end
 
+"""
+    set_normalized_coefficients(
+        con_ref::ConstraintRef,
+        variable,
+        new_coefficients::Vector{Tuple{Int64,T}},
+    )
+
+Set the coefficients of `variable` in the constraint `constraint` to
```
Comment (odow) [inline src/constraints.jl:692]:
```suggestion
Set the coefficients of `variable` in the constraint `con_ref` to
```

File: src/constraints.jl [hunk 2]
```diff
@@ -682,6 +682,46 @@ function set_normalized_coefficient(
...
+        con_ref::ConstraintRef,
+        variable,
+        new_coefficients::Vector{Tuple{Int64,T}},
+    )
+
+Set the coefficients of `variable` in the constraint `con_ref` to
+`new_coefficients`.
+
+Note that prior to this step, JuMP will aggregate multiple terms containing the
+same variable.
```
Comment (joaquimg) [inline src/constraints.jl:696]:
```suggestion
Note that prior to this step, during constraint creation, JuMP will aggregate
multiple terms containing the same variable.
```

**Review by joaquimg — APPROVED** [review]

File: src/constraints.jl [hunk 3]
```diff
@@ -682,6 +682,46 @@ function set_normalized_coefficient(
...
+```jldoctest; setup = :(using JuMP), filter=r"≤|<="
+model = Model()
+@variable(model, x)
+@constraint(model, con, [2x + 3x, 4x] in MOI.Nonnegatives(2))
+set_normalized_coefficients(con, x, [(1, 2.0), (2, 5.0)])
+con
+
+# output
+
+con : [2 x, 5x] ∈ MathOptInterface.Nonnegatives(2)
```
Comment (blegat) [inline src/constraints.jl:707]: This printing is weird, why is there a space after 2 and not 5

Comment (odow) [inline src/constraints.jl:707]:
```suggestion
con : [2 x, 5 x] ∈ MathOptInterface.Nonnegatives(2)
```

Comment (odow) [reply src/constraints.jl]: I think that was just my typo. Doc tests didn't seem to fail on it?

File: src/constraints.jl [hunk 4]
```diff
@@ -682,6 +682,46 @@ function set_normalized_coefficient(
...
+`new_coefficients`.
+
+Note that prior to this step, during constraint creation, JuMP will aggregate
+multiple terms containing the same variable.
+
+```jldoctest; setup = :(using JuMP), filter=r"≤|<="
+model = Model()
+@variable(model, x)
+@constraint(model, con, [2x + 3x, 4x] in MOI.Nonnegatives(2))
+set_normalized_coefficients(con, x, [(1, 2.0), (2, 5.0)])
```
Comment (odow) [inline src/constraints.jl:702]: One design decision is whether this is the nicest design. Another option might be `[2.0, 5.0]`, and you have to provide explicit zeros.

Comment (odow) [reply src/constraints.jl:703]: Let's keep close to the MOI input.

**Review by blegat — APPROVED** [review]

**Pull request merged**